modules/custom/az_publication/az_publication.module

```diff
@@ -306,12 +346,22 @@ function az_publication_node_view(array &$build, EntityInterface $entity, Entity
           "author" => $renderlink,
           "csl-entry" => function ($cslItem, $renderedText) {
             // Remove citation number tag.
-            // In general it's never a good idea to parse html with regex.
-            // This works ONLY because the output is a known pattern.
             if (!empty($cslItem->additionalLinks)) {
               $renderedText .= ' ' . $cslItem->additionalLinks;
             }
-            return preg_replace('#<div class="csl-left-margin">(.*?)</div>#', '', $renderedText);
+            $dom = new DOMDocument();
```

Comment:
Does switching to DOM-based parsing here resolve an issue with the current regex-based parsing?

Reply:
I'm not sure anymore, but line 309 says not to parse HTML with regex.